src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1174,7 +1174,14 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       } else tree.tpe match {
         case atp @ AnnotatedType(_, _) if canAdaptAnnotations(tree, this, mode, pt) => // (-1)
           adaptAnnotations(tree, this, mode, pt)
-        case ct @ ConstantType(value) if mode.inNone(TYPEmode | FUNmode) && (ct <:< pt) && canAdaptConstantTypeToLiteral => // (0)
+        case ct @ ConstantType(value) if mode.inNone(TYPEmode | FUNmode | PATTERNmode) && (ct <:< pt) && canAdaptConstantTypeToLiteral => // (0)
+          /* don't adapt in PATTERNmode only because of the duplicated alternatives
```

Comment:
Thanks! I'll find out why we don't adapt in `TYPEmode` and `FUNmode` and expand the comment, if you'd like. The `PATTERNmode` case seemed like it would be unintuitive without more context, so that's why I put it in.